core/src/main/java/org/opensearch/remote/metadata/client/SdkClient.java

```diff
@@ -108,7 +108,7 @@ public CompletionStage<GetDataObjectResponse> getDataObjectAsync(GetDataObjectRe
      * @return A completion stage encapsulating the response or exception
      */
     public CompletionStage<GetDataObjectResponse> getDataObjectAsync(GetDataObjectRequest request) {
-        validateTenantId(request.tenantId());
+        validateTenantIdNotNull(request.tenantId());
```

Comment:
Add this to the list of things that I really want explained in great detail why it is necessary.  I already suggested coce to solve the problem we're trying to solve without changing core assumptions for this class.

Reply:
I'm not following, do you mean to add similar method for bulk?

Reply:
Sorry, I think I was still thinking we were dealing with the null parser discussion.  But I do still think having an entire method for what amounts to a null check, when such a check already exists, isn't the right approach.

Resolving this comment and addressing this in the context of the unneeded method.